### PR TITLE
util: Bump GPUFS build docker to 5.4.2

### PR DESCRIPTION
**Pull request by abmerop**
This dockerfile is used to *build* applications (e.g., from gem5-resources) which can be run using full system mode in a GPU build. The next releases disk image will use ROCm 5.4.2, therefore bump the version from 4.2 to that version.

Again this is used to *build* input applications only and is not needed to run or compile gem5 with GPUFS. For example:

$ docker build -t rocm54-build .
/some/gem5-resources/src/gpu/lulesh$ docker run --rm -u $UID:$GID -v \
    ${PWD}:${PWD} -w ${PWD} rocm54-build make

Change-Id: If169c8d433afb3044f9b88e883ff3bb2f4bc70d2

### CODE REVIEW

File: util/dockerfiles/gpu-fs/Dockerfile [hunk 1]
```diff
@@ -27,7 +27,7 @@
 # ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 # POSSIBILITY OF SUCH DAMAGE.
 
-FROM --platform=${BUILDPLATFORM} ubuntu:20.04
+FROM ubuntu:20.04
```
Comment (powerjg) [inline util/dockerfiles/gpu-fs/Dockerfile]: Why this change? The old version would, in theory, allow this to build Arm binaries as well as x86 binaries.

I know that it's unlikely that someone would want to build Arm binaries for the AMD GPU, but it doesn't hurt to have the old version, from what I understand.

Comment (abmerop) [reply util/dockerfiles/gpu-fs/Dockerfile]: Ah ok.  I removed it because I didn't know what it does, but I'll add it back

**Review by powerjg — COMMENTED** [review]
lgtm. Thanks for the update. I'll let Bobby sign off so he can rebuild and push the new image.

**Review by mattsinc — APPROVED** [review]

**Comment by abmerop** [comment]
Thanks Jason. FYI will be my last GPUFS related patch before v23.1.  Although I may write some documentation stuff on the other repos :)

**Comment by powerjg** [comment]
@BobbyRBruce want to make sure you see this to rebuild the image

**Comment by BobbyRBruce** [comment]
> @BobbyRBruce want to make sure you see this to rebuild the image

It's rebuilt and uploaded.
